### DISCUSSION TITLE
What does the Size on <Figure> do?

### DISCUSSION THREAD

**Discussion by sturlath** [Megabit/Blazorise#2464] [Q&A] [answered]
Am I misunderstanding how ` Size`  on `<Figure>` works? It provides a frame but images bigger than it expand beyond it. 

If I have a bigger image than this 256x256 it goes beyond that range.

the code
```xml
<Row>
    <Column>
        <Fields>
            <Field ColumnSize="ColumnSize.Is12.OnDesktop">  
                <Figure Size="FigureSize.Is256x256">
                    <FigureImage Source="@Image" />
                </Figure>
            </Field>
            <Field ColumnSize="ColumnSize.Is12.OnDesktop">
                <FileEdit Filter="image/*" Changed="@OnChanged" />      
            </Field>
        </Fields>
    </Column>
</Row>
```

like this, where the figure is 256x256 but the image goes below

![image](https://user-images.githubusercontent.com/2386572/121506381-2b7fa480-c9d3-11eb-8752-063738f0f6f9.png)

**BUT**
It works if I add  `Style=height:256px;width:256px`  to `<FigureImage>` 

like this

```xml
<Figure Size="FigureSize.Is256x256">
    <FigureImage Source="@Image"  Style="height:256px;width:256px;object-fit:contain;" />
</Figure>
```

Sorry for these questions of mine but my CSS foo is not strong at all, but I try.. :-)

**Comment by stsrki** [comment, marked as answer]
It works by setting the Figure **container** to a set size, in pixels. I believe you still need to adjust the actual image size. Like you already did.

**Reply by sturlath** [reply]
I had tried earlier to set the size on Figure like this

```xml
<Figure Size="FigureSize.Is256x256" Style="width: 256px; height: 256px">
```

but that didn´t do anything.. I still had to add  `Style="width: 256px; height: 256px"` to `<FigureImage>` also.

I still don't understand what FigureSize is supposed to be doing but **that´s OK**! I got this to work so just ploughing ahead :-)
